src/index.ml

```diff
@@ -430,6 +430,7 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
   let merge ~witness t =
     Log.debug (fun l -> l "merge %S" t.root);
     let tmp_path = t.root // "tmp" // "index" in
+    IO.wait_for_lock tmp_path;
```

Comment:
I am not sure to understand why do we wait for the path to disappear?

Would probably be cleaner to use `flock` like @mefyl did on [dune cache](https://github.com/ocaml/dune/pull/2493/commits/7cf3776981ff0c965856793130705892f3b5d8db) to ensure there is only 1 writer process.